packages/fxa-settings/src/pages/Signup/container.spec.tsx

```diff
@@ -0,0 +1,439 @@
+ *  5. Assert on mocked functions and spies
+ */
+
+import React from 'react';
+
+// TIP - Import modules for mocking. Not that `* as` lend themselves to using jest.spyOn.
+import * as SignupModule from './index';
+import * as ModelsModule from '../../models';
+import * as LoadingSpinnerModule from 'fxa-react/components/LoadingSpinner';
+import * as ApolloModule from '@apollo/client';
+import * as UseValidateModule from '../../lib/hooks/useValidate';
+import * as FirefoxModule from '../../lib/channels/firefox';
+import * as CryptoModule from 'fxa-auth-client/lib/crypto';
+import * as ReachRouterModule from '@reach/router';
+import * as ReactUtils from 'fxa-react/lib/utils';
+
+// Typical imports
+import { renderWithLocalizationProvider } from 'fxa-react/lib/test-utils/localizationProvider';
+import { screen } from '@testing-library/react';
+import SingupContainer, { SignupContainerIntegration } from './container';
+import { IntegrationType, useAuthClient } from '../../models';
+import { MozServices } from '../../lib/types';
+import { FirefoxCommand } from '../../lib/channels/firefox';
+import { Constants } from '../../lib/constants';
+import { SignupProps } from './interfaces';
+import { AuthUiErrors } from '../../lib/auth-errors/auth-errors';
+import { GraphQLError } from 'graphql';
+import { ApolloClient, ApolloError } from '@apollo/client';
+import { ModelDataProvider } from '../../lib/model-data';
+import AuthClient from 'fxa-auth-client/browser';
+
+// TIP - Sometimes, we want to mock inputs. In this case they can be mocked directly and
+// often times a mocking util isn't even necessary. Note that using the Dependency Inversion
+// Principle, will limit the amount you need to mock. Here, we let the container specify
+// the SignupContainerIntegration interface which is a subset (and therefore compatible)
+// larger and often more specific interfaces like Integration, OAuthIntegration, etc...
+let integration: SignupContainerIntegration;
+function mockIntegration() {
+  integration = {
+    type: IntegrationType.SyncDesktop,
+    getService: () => MozServices.Default,
+    features: {
+      allowUidChange: false,
+      fxaStatus: false,
+      handleSignedInNotification: false,
+      reuseExistingSession: false,
+      supportsPairing: false,
+      webChannelSupport: false,
+    },
+  };
+}
+let serviceName: MozServices;
+function mockServiceName() {
+  serviceName = MozServices.Default;
+}
+
+// TIP - Note that we don't actually want to render the Signup or LoadingSpinner react components,
+// since we aren't testing their logic. There fore it's fine to just return something super
+// simple. And check assumptions about props if necessary.
+let currentSignupProps: SignupProps | undefined;
+function mockSignupModule() {
+  currentSignupProps = undefined;
+  jest
+    .spyOn(SignupModule, 'Signup')
+    .mockImplementation((props: SignupProps) => {
+      currentSignupProps = props;
+      return <div>sign up mock</div>;
```

Comment:
Just a nit, maybe we use `Signup mock` and `LoadingSpinner mock` text just to follow the component name.